rclpy/rclpy/executors.py

```diff
@@ -802,10 +802,13 @@ def _spin_once_impl(
         else:
             self._executor.submit(handler)
             self._futures.append(handler)
-            for future in self._futures:  # check for any exceptions
+            self._futures.append(handler)
```

Comment:
This looks like a typo; we shouldn't append the handler to tlist of `futures` twice, I don't think.

Reply:
Yes, you are right! Sorry. Also suspicious that the tests did not catch this ...

```suggestion
```